pylint_django/tests/input/func_unused_arguments.py

```diff
@@ -0,0 +1,26 @@
+"""
+Checks that Pylint still complains about unused-arguments if a function/method
+contains an argument named `request`.
+"""
+# pylint: disable=missing-docstring
+
+from django.http import JsonResponse
+from django.views import View
+
+# Pylint generates the warning `redefined-outer-name` if an argument name shadows
+# a variable name from an outer scope. But if that argument name is ignored this
+# warning will not be generated.
+# Therefore define request here to cover this behaviour in this test case.
+
+request = None  # pylint: disable=invalid-name
+
+
+def user_detail(request, user_id):  # [unused-argument]
```

Comment:
Shound't you get a `redefined-outer-scope` for request here ?

I am +1 for monkey patching only the private method if that is the minimum amount of changes and doesn't break existing functionality .

Reply:
The more I think about this the more I agree. But this is the current behaviour of pylint (see comment above).